src/components/DeleteProduct.tsx

```diff
@@ -1,38 +1,57 @@
 'use client';
 import { useState } from 'react';
-import { doc, deleteDoc } from 'firebase/firestore';
-import { db, collections } from '@/utils/firebase';
 
 interface DeleteProductProps {
   productId: string;
   onDeleted?: () => void;
+  activeId: string | null;
+  setActiveId: (id: string | null) => void;
 }
 
-export default function DeleteProduct({ productId, onDeleted }: DeleteProductProps) {
-  const [showConfirm, setShowConfirm] = useState(false);
+export default function DeleteProduct({
+  productId,
+  onDeleted,
+  activeId,
+  setActiveId,
+}: DeleteProductProps) {
+  const isActive = activeId === productId;
   const [showSuccess, setShowSuccess] = useState(false);
 
   const handleDelete = async () => {
-    await deleteDoc(doc(db, collections.products, productId));
-    setShowConfirm(false);
-    setShowSuccess(true);
-    if (onDeleted) onDeleted();
-    setTimeout(() => {
-      setShowSuccess(false);
-    }, 2000);
+    try {
+      const res = await fetch('/api/delete-product', {
```

Comment:
When you define an API endpoint like /api/delete-product in NextJS, it will accept any incoming requests from anyone on the internet. It should not be used without proper authentication. A common way to do it is to pass a token that the user gets when they sign in as:
```
{
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
      ody: JSON.stringify({ productId }),
    }
```

Reply:
Here, you can avoid this issue by using Server Action instead - then you don't need to provide the token or secure the implementation in any similar way.